### PR TITLE
Add New Site: fix JP card autofocus

### PR DESCRIPTION
**Pull request by keoshi**
This fixes some weirdness with the input autofocus in the Jetpack card on Add New Site. This was reported in p7kMJG-4nr-p2

This PR introduces an autoFocus prop in the `site-url-input` component, which allows us to target it differently depending on the view it’s presented. This is enabled by default, but disabled in this specific “Add New Site” view.

---

**Heads-up**: I had to add a `--no-verify` flag before I was able to commit. This was the thrown error:

```
/client/jetpack-connect/site-url-input.jsx
  124:7  error  The autoFocus prop should not be used, as it can reduce usability and accessibility for users  jsx-a11y/no-autofocus
```

### Before

image

image

### After

image

image

### Testing

- Go to http://calypso.localhost:3000/jetpack/new?ref=calypso-selector
- Ensure the input for a Jetpack URL is not autofocused

### CODE REVIEW

File: client/jetpack-connect/site-url-input.jsx [hunk 1]
```diff
@@ -119,7 +121,7 @@ class JetpackConnectSiteUrlInput extends PureComponent {
 						ref={ this.refInput }
 						id="siteUrl"
 						autoCapitalize="off"
-						autoFocus="autofocus"
```
Comment (omidfi) [inline client/jetpack-connect/site-url-input.jsx]: Hi @keoshi,

I get the same effect with just removing line 122 

```html
autoFocus="autofocus"
```

Is there a place where autoFocus is really needed for this component? I only could find site-url-input used in client/jetpack-connect/main.jsx and couldn't see why autoFocus would be needed in there.

I hope my comment is not way too off, as I'm new to the codebase :)

Comment (tyxla) [reply client/jetpack-connect/site-url-input.jsx]: @omidfi, yes, autofocusing this field is expected and needed in the Jetpack Connect flow, for example in https://wordpress.com/jetpack/connect/

**Comment by ockham** [comment]
> **Heads-up**: I had to add a `--no-verify` flag before I was able to commit. This was the thrown error:
> 
> ```
> /client/jetpack-connect/site-url-input.jsx
>   124:7  error  The autoFocus prop should not be used, as it can reduce usability and accessibility for users  jsx-a11y/no-autofocus
> ```


If we _are_ going to purposefully ignore the a11y lint error permanently in this case, we should add an inline `// eslint-disable-line jsx-a11y/no-autofocus` pragma. Otherwise anyone ever touching the same lines will also have to use the `--no-verify` flag, and CircleCI will continue to flag it as an error.

**Comment by keoshi** [comment]
@ockham @tyxla What do you guys recommend doing here? The usability concern is not really valid here because we're explicitly declaring a non-autofocused input, not the other way around.

Bernie, the solution you propose would like something like this if I'm interpreting that right?

`autoFocus={ autoFocus } // eslint-disable-line jsx-a11y/no-autofocus`

**Comment by tyxla** [comment]
Having the `eslint` exception here makes sense to me.

**Comment by ockham** [comment]
> Bernie, the solution you propose would look something like this if I'm interpreting that right?

> `autoFocus={ autoFocus } // eslint-disable-line jsx-a11y/no-autofocus`

@keoshi Yeah, exactly!

**Comment by keoshi** [comment]
Just pushed that change. Thanks a bunch @ockham and @tyxla ! 🙌

**Pull request merged**